fsx/main.tf

```diff
@@ -0,0 +1,327 @@
+provider "aws" {
+  default_tags {
+    tags = {
+      Owner       = var.owner_tag
+      Environment = var.environment_tag
+    }
+  }
+}
+
+provider "cloudinit" {}
+
+data "aws_availability_zones" "available" {}
+
+/**************************************************************************************************
+ * Networking
+ *************************************************************************************************/
+
+resource "aws_vpc" "vpc" {
+  cidr_block           = "10.0.0.0/16"
+  enable_dns_hostnames = true
+}
+
+# For simplicity: one public subnet in the desired availability zone. A private subnet with NAT
+# would work too.
+resource "aws_subnet" "public" {
+  vpc_id                  = aws_vpc.vpc.id
+  cidr_block              = "10.0.0.0/20"
+  availability_zone       = var.availability_zone
+  map_public_ip_on_launch = true
+}
+
+resource "aws_internet_gateway" "igw" {
+  vpc_id = aws_vpc.vpc.id
+}
+
+resource "aws_route_table" "public" {
+  vpc_id = aws_vpc.vpc.id
+  route {
+    cidr_block = "0.0.0.0/0"
+    gateway_id = aws_internet_gateway.igw.id
+  }
+}
+
+resource "aws_route_table_association" "public" {
+  route_table_id = aws_route_table.public.id
+  subnet_id      = aws_subnet.public.id
+}
+
+# Umbrella security group for Batch compute environments & filesystem mount targets, allowing any
+# traffic within the VPC and outbound-only Internet access.
+# The ingress could be locked down to allow only FSxL traffic within the VPC,
+# https://docs.aws.amazon.com/fsx/latest/LustreGuide/limit-access-security-groups.html#fsx-vpc-security-groups
+resource "aws_security_group" "all" {
+  name   = var.environment_tag
+  vpc_id = aws_vpc.vpc.id
+  ingress {
+    from_port   = 0
+    to_port     = 0
+    protocol    = "-1"
+    cidr_blocks = [aws_vpc.vpc.cidr_block]
+  }
+  # Uncomment to open SSH access via EC2 Instance Connect
+  /*
+  ingress {
+    from_port   = 22
+    to_port     = 22
+    protocol    = "tcp"
+    cidr_blocks = ["0.0.0.0/0"]
+  }
+  */
+  egress {
+    from_port   = 0
+    to_port     = 0
+    protocol    = "-1"
+    cidr_blocks = ["0.0.0.0/0"]
+  }
+}
+
+/**************************************************************************************************
+ * FSxL
+ *************************************************************************************************/
+
+resource "aws_fsx_lustre_file_system" "lustre" {
+  subnet_ids                    = [aws_subnet.public.id]
```

Comment:
Does this put the volume in a public subnet? Is there any reason it can't be private?

Reply:
It's just more moving parts (trying to keep this config as a minimal starting point), but I'll look into it